apps/api/src/app/workflows-v2/usecases/post-process-workflow-update/post-process-workflow-update.usecase.ts

```diff
@@ -82,18 +110,185 @@ export class PostProcessWorkflowUpdate {
   private hasBodyIssues(issue: StepIssues) {
     return issue.body && Object.keys(issue.body).length > 0;
   }
-  private validateSteps(steps: NotificationStepEntity[]): Record<string, StepIssuesDto> {
+
+  private async validateSteps(
+    steps: NotificationStepEntity[],
+    _workflowId: string,
+    user: UserSessionData
+  ): Promise<Record<string, StepIssuesDto>> {
     const stepIdToIssues: Record<string, StepIssuesDto> = {};
-    for (const step of steps) {
-      stepIdToIssues[step._templateId] = {
-        body: this.addStepBodyIssues(step),
-        controls: step.issues?.controls,
-      };
-    }
+
+    await Promise.all(
+      steps.map(async (step) => {
+        stepIdToIssues[step._templateId] = {
+          body: this.addStepBodyIssues(step),
+          controls: await this.buildControlIssues(step, _workflowId, user),
+        };
+      })
+    );
 
     return stepIdToIssues;
   }
 
+  private async buildControlIssues(step: NotificationStepEntity, _workflowId: string, user: UserSessionData) {
```

Comment:
Currently, there's no reliable way to determine whether a control value should be validated or not.

For example, we might have unit/amount control values in the code-first value. Should we validate them the same way or not?